### PR TITLE
Database and 2 deep recommendations

### PR DESCRIPTION
**Pull request by KCraw**
I did 2 things here:

(1) The mess in App was annoying me, so I made a lightweight in-memory database to house data and added the concept of a 'user' to that database. No more loading raw JSON. I did make a slight change to data, so that now it's recommendation.event and recommendation.place instead of recommendation.event.place. 

(2) I changed the RecommendationsScene to only ever be 2 cards deep to make the usage more intuitive. If we went with this, we'd pass in a nextRecommendation object as a prop, and keep a currentRecommendation object as a state. That's really all the scene was doing before, so why should we pass in a whole stack? 

This is based on the premise that database functionality (including optimistic updates) should determine what the new stack of recommendations looks like (even if just removing the just saved or dismissed recommendation from the list), and that a user or other top level object will be responsible for determining how deep a stack of recommendations to fetch for network traffic purposes. 

If there was no affinity algorithm or AI algorithm it wouldn't be necessary at all to have the component keep track of the stack. We would optimistically update the recommendations to no longer include the dismissed/saved recommendation, update the recommendations and the top recommendation passed into the scene, batch a query to the server, and when the server updates the store it should not trigger a rerender since the server data would match the optimistic update. 

The only reason this is necessary for a UI component is because the server data often will not match the optimistic update, since the whole set of recommendations might have changed, including the top recommendation, once the algorithms run (possibly, maybe, since we haven't built the backend yet). When that happens, the UI will still display currentRecommendation, and swiping it will trigger another optimistic update followed by a server query if currentRecommendation is contained in the recommendation stack, or just the server query if not, starting the process all over again. 

The whole thing should be impossible to get out of sync, which I'm afraid might happen if we are passing in and swiping through a whole stack at the component UI level. And as long as the user batches enough recommendations for network issues, we shouldn't run afoul of those concerns. 

Like I said, we were already doing this anyway by only ever looking at the top of the deck. This just makes it explicit and less confusing conceptually when we come back to it in 6 months or bring on other developers.

### CODE REVIEW

File: components/app/index.js [hunk 1]
```diff
@@ -27,56 +27,44 @@ export default class Gustave extends Component {
     name: 'Recommendations',
   };
 
-  // Temporary to simulate mutation + optimistic update
+  // Temporary for demo only
```
Comment (jimbol) [inline components/app/index.js]: Faking a load, good for development too

**Comment by KCraw** [comment]
Remove button removes from saved and also adds to dismissed...

I'm not sure that's how it will really be handled, but it was weird to save a recommendation, change my mind, and then have it suggested to me right away again. So for now treating a removal from saved the same as a dismissal.

**Comment by KCraw** [comment]
PR ready for review

**Comment by jimbol** [comment]
Closing this since the other ticket has all these changes.

**Pull request closed without merging**